app/src/main/java/com/keylesspalace/tusky/db/TimelineDao.kt

```diff
@@ -113,8 +113,7 @@ AND
             muted = status.muted,
             pinned = status.pinned ?: false,
             card = gson.toJson(status.card),
-            language = status.language,
-            filtered = status.filtered
+            language = status.language
```

Comment:
Hm, how does that solve the problem?

(How is "filtered" then stored?)

Reply:
Filtered contains the filters that apply to the post. When the post is revealed, we set it to null. On interaction (boost, fav, etc) it would set the filters again. Not updating the filters here means the posts stays revealed.

Reply:
Should it also / still store the opened state once you have expanded something?

I tested this with:

- Expand a filtered post
- Fav it (ok, nothing bad happens)
- Reload timeline: the post is again filtered (collapsed)

Reply:
Yes it probably should. But looks like that would require a larger refactoring 🫣

Reply:
I think we should merge this as it improves the situation and include it in the 24 release, and I will do the refactor later.

Reply:
@Lakoja in my opinion the post *should* recollapse on timeline reload, just not on Interaction. I for example have a filter for Doctor Who to avoid spoilers, and on more than one occasion I forgot on Saturday there's a new episode up and opened the filter thinking it refers to the last one I saw. I would very much want the filter to recollapse when I reload the timeline in panic trying to avoid reading the post I've accidentally revealed.
maybe that's too extreme a position (I know recollapsing on reload doesn't happen for CWs for example) but I'd argue it justified. but then again, it's all matter of opinion.